tensorflow_addons/image/interpolate_spline_test.py

```diff
@@ -0,0 +1,365 @@
+        value_dim = train_values.shape[-1]
+        train_points_ph = tf1.placeholder(
+            dtype=train_points.dtype, shape=[None, None, feature_dim])
+        train_values_ph = tf1.placeholder(
+            dtype=train_values.dtype, shape=[None, None, value_dim])
+        query_points_ph = tf1.placeholder(
+            dtype=query_points.dtype, shape=[None, None, feature_dim])
+
+        order = 1
+        reg_weight = 0.01
+
+        interpolator = interpolate_spline(train_points_ph, train_values_ph,
+                                          query_points_ph, order, reg_weight)
+
+        target_interpolation = tp.HARDCODED_QUERY_VALUES[(order, reg_weight)]
+        target_interpolation = np.array(target_interpolation)
+        with self.cached_session() as sess:
+
+            (train_points_value, train_values_value,
+             query_points_value) = sess.run(
+                 [train_points, train_values, query_points])
+
+            interp_val = sess.run(
+                interpolator,
+                feed_dict={
+                    train_points_ph: train_points_value,
+                    train_values_ph: train_values_value,
+                    query_points_ph: query_points_value
+                })
+            self.assertAllClose(interp_val[0, :, 0], target_interpolation)
+
+    def test_fully_unspecified_shape(self):
+        """Ensure that erreor is thrown when input/output dim unspecified."""
+        self.skipTest("TODO: port to tf2.0 / eager")
+        tp = _QuadraticPlusSinProblemND()
+        (query_points, _, train_points,
+         train_values) = tp.get_problem(dtype='float64')
+
+        # Construct placeholders such that the batch size, number of train points,
+        # and number of query points are not known at graph construction time.
+        feature_dim = query_points.shape[-1]
+        value_dim = train_values.shape[-1]
+        train_points_ph = tf1.placeholder(
+            dtype=train_points.dtype, shape=[None, None, feature_dim])
+        train_points_ph_invalid = tf1.placeholder(
+            dtype=train_points.dtype, shape=[None, None, None])
+        train_values_ph = tf1.placeholder(
+            dtype=train_values.dtype, shape=[None, None, value_dim])
+        train_values_ph_invalid = tf1.placeholder(
+            dtype=train_values.dtype, shape=[None, None, None])
+        query_points_ph = tf1.placeholder(
+            dtype=query_points.dtype, shape=[None, None, feature_dim])
+
+        order = 1
+        reg_weight = 0.01
+
+        with self.assertRaises(ValueError):
+            _ = interpolate_spline(train_points_ph_invalid, train_values_ph,
+                                   query_points_ph, order, reg_weight)
+
+        with self.assertRaises(ValueError):
+            _ = interpolate_spline(train_points_ph, train_values_ph_invalid,
+                                   query_points_ph, order, reg_weight)
+
+    def test_interpolation_gradient(self):
```

Comment:
I wonder if [tf.test.compute_gradient](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/test/compute_gradient) can help you.